File: Obligatorio1DA2/Domain/Reservation.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;

namespace Domain
{
    public class Reservation
    {
        public int Id { get; set; }
        public Guid Code { get; set; }
        public Lodging Lodging { get; set; }
        public Client Client { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
        public State State {get; set;}
        public string StateDescription { get; set; }
        public double Price { get; set; }

        public Reservation()
        {

        }
        public override bool Equals(object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (!(obj is Reservation))
            {
                return false;
            }
            return Code == ((Reservation)obj).Code;
        }
    }
}

File: Codigo/UruguayNatural/src/Models/Reservation.ts
export class Reservation {
    Code: number;
    LodgingId: number;
    ClientId: number;
    Checkin: string;
    Checkout: string;
    StateId: number;
    StateDescription: string;
    Price: number;

    constructor(Code: number, LodgingId: number, ClientId: number, Checkin: string,
        Checkout: string, StateId: number, StateDescription: string, Price: number) {
        this.Code = Code;
        this.LodgingId = LodgingId;
        this.ClientId = ClientId;
        this.Checkin = Checkin;
        this.Checkout = Checkout;
        this.StateId = StateId;
        this.StateDescription = StateDescription;
        this.Price = Price;
    }

}
